src/query.c

```diff
@@ -508,69 +537,20 @@ static IndexIterator *iterateExpandedTerms(QueryEvalCtx *q, Trie *terms, const c
   IndexIterator **its = rm_calloc(itsCap, sizeof(*its));
 
   rune *rstr = NULL;
+  size_t tok_len = 0;
   t_len slen = 0;
   float score = 0;
   int dist = 0;
 
   // an upper limit on the number of expansions is enforced to avoid stuff like "*"
   while (TrieIterator_Next(it, &rstr, &slen, NULL, &score, &dist) &&
          (itsSz < q->config->maxPrefixExpansions)) {
-
-    // Create a token for the reader
-    RSToken tok = (RSToken){
-        .expanded = 0,
-        .flags = 0,
-        .len = 0,
-    };
-    tok.str = runesToStr(rstr, slen, &tok.len);
-    if (q->sctx && q->sctx->redisCtx) {
-      RedisModule_Log(q->sctx->redisCtx, "debug", "Found fuzzy expansion: %s %f", tok.str, score);
-    }
-
-    RSQueryTerm *term = NewQueryTerm(&tok, q->tokenId++);
-
-    // Open an index reader
-    IndexReader *ir = Redis_OpenReader(q->sctx, term, &q->sctx->spec->docs, 0,
-                                       q->opts->fieldmask & opts->fieldMask, q->conc, 1);
-
-    rm_free(tok.str);
-    if (!ir) {
-      Term_Free(term);
-      continue;
-    }
-
-    // Add the reader to the iterator array
-    its[itsSz++] = NewReadIterator(ir);
-    if (itsSz == itsCap) {
-      itsCap *= 2;
-      its = rm_realloc(its, itsCap * sizeof(*its));
-    }
+    addTerm(runesToStr(rstr, slen, &tok_len), tok_len, q, &its, &itsSz, &itsCap, opts);
```

Comment:
This is tricky behavior.
`runesToStr` computes tok_len which you then pass as an argument.
Maybe you can pass a `RSToken` as a parameter to addTerm instead?

It will feel a bit cleaner I think.

Reply:
We actually do this in more places and generally it works fine of-course, but I see what you mean we can do this slower (which will also be a bit cleaner)